src/plugins/hiddenColumns/hiddenColumns.js

```diff
@@ -150,41 +135,45 @@ class HiddenColumns extends BasePlugin {
    * Disables the plugin functionality for this Handsontable instance.
    */
   disablePlugin() {
+    this.hot.columnIndexMapper.unregisterMap(PLUGIN_NAME);
     this.settings = {};
-    this.hiddenColumns = [];
-    this.lastSelectedColumn = -1;
 
-    this.hot.render();
     super.disablePlugin();
     this.resetCellsMeta();
   }
 
   /**
    * Shows the provided columns.
    *
-   * @param {number[]} columns Array of column indexes.
+   * @param {number[]} columns Array of visual column indexes.
    */
   showColumns(columns) {
-    const currentHideConfig = this.hiddenColumns;
-    const validColumns = this.isColumnDataValid(columns);
+    const currentHideConfig = this.getHiddenColumns();
+    const isConfigValid = this.isValidConfig(columns);
     let destinationHideConfig = currentHideConfig;
 
-    if (validColumns) {
-      destinationHideConfig = this.hiddenColumns.filter(hiddenColumn => columns.includes(hiddenColumn) === false);
+    if (isConfigValid) {
+      destinationHideConfig = currentHideConfig.filter(column => !columns.includes(column));
```

Comment:
The `getHiddenColumns` should return the list with unique indexes and `Set` would look good for that purpose. However, the function should return simple, native value if only because the `Set` is the feature from the `ES6`. We still support some browsers not supporting the `ES2015`.

If the used `currentHideConfig` contains only unique values inside an array for me it's useless to cast it to the `Set`.

Reply:
@pnowak Similar helper situation here. Maybe we could check if we can introduce `arrayIntersection` and `arrayDifference` helpers. Maybe we can compare the performance and choose the best solution?

Reply:
We discussed the fact, that it may be used for performance optimization. It should be checked within another PR (helper `arrayUnique` will be changed probably).